apps/devhub/tests/test_views.py

```diff
@@ -756,26 +756,50 @@ def setUp(self):
         super(TestDelete, self).setUp()
         self.get_addon = lambda: Addon.objects.filter(id=3615)
         assert self.client.login(username='del@icio.us', password='password')
+        self.user = UserProfile.objects.get(email='del@icio.us')
         self.get_url = lambda: self.get_addon()[0].get_dev_url('delete')
 
+    def make_theme(self):
+        theme = addon_factory(
+            name='xpi name', type=amo.ADDON_PERSONA, slug='theme-slug')
+        theme.authors.through.objects.create(addon=theme, user=self.user)
+        return theme
+
     def test_post_not(self):
         r = self.client.post(self.get_url(), follow=True)
         eq_(pq(r.content)('.notification-box').text(),
-            'Password was incorrect. Add-on was not deleted.')
+            'URL name was incorrect. Add-on was not deleted.')
         eq_(self.get_addon().exists(), True)
 
     def test_post(self):
-        r = self.client.post(self.get_url(), {'password': 'password'},
+        self.get_addon().get().update(slug='addon-slug')
+        r = self.client.post(self.get_url(), {'slug': 'addon-slug'},
                              follow=True)
         eq_(pq(r.content)('.notification-box').text(), 'Add-on deleted.')
         eq_(self.get_addon().exists(), False)
 
-    def test_post_theme(self):
-        Addon.objects.get(id=3615).update(type=amo.ADDON_PERSONA)
-        r = self.client.post(self.get_url(), {'password': 'password'},
+    def test_post_wrong_slug(self):
+        self.get_addon().get().update(slug='addon-slug')
+        r = self.client.post(self.get_url(), {'slug': 'theme-slug'},
                              follow=True)
+        eq_(pq(r.content)('.notification-box').text(),
+            'URL name was incorrect. Add-on was not deleted.')
+        eq_(self.get_addon().exists(), True)
+
+    def test_post_theme(self):
+        theme = self.make_theme()
+        r = self.client.post(
+            theme.get_dev_url('delete'), {'slug': 'theme-slug'}, follow=True)
```

Comment:
`self.get_url()` is for the add-on from the fixture. It was previously just changing the type and only tested the negative case but you can't actually delete an add-on that is pretending to be a theme. It raises an error about not having a `Persona`.

I updated the method to take the add-on and then realised that I could just call a method on the add-on itself and it didn't really make sense anymore.